clang/lib/CIR/CodeGen/CIRGenOpenACCClause.h

```diff
@@ -82,6 +83,17 @@ class OpenACCClauseCIREmitter final
     return conversionOp.getResult(0);
   }
 
+  mlir::Value createConstantInt(mlir::Location loc, unsigned width,
+                                int64_t value) {
+    mlir::IntegerType ty = mlir::IntegerType::get(
+        &cgf.getMLIRContext(), width,
+        mlir::IntegerType::SignednessSemantics::Signless);
+    auto constOp = builder.create<mlir::arith::ConstantOp>(
```

Comment:
Any reason why this is `arith` dialect constant not `cir` constant?

Reply:
At the moment it looks like we're pretty close to the end of what I'll have to do here.  There are a couple of clauses that just take ints, so we do those conversions above, another constant or two, but they are pretty rote?  I don't know if the work for a pass to just do a couple of type conversions would be particularly beneficial.

BUT it is definitely something I want to keep  in mind.

Reply:
In the long run, it might be worthwhile to introduce `IntegerTypeInterface` to core dialects so that dialect like `CIR` it its types can interoperate.